extras/ci/analytics/ci_visualization.py

```
@@ -0,0 +1,970 @@
+import os
+import sys
+from collections import defaultdict
+
+# Timeline constants
+PIXELS_PER_HOUR = 265
+ROW_HEIGHT = 28
+TOTAL_WIDTH = PIXELS_PER_HOUR * 24  # 6360px for 24h
+
+
+def parse_args():
+    script_dir = os.path.dirname(os.path.abspath(__file__))
```

Ruff (0.15.2)

[error] Local variable `script_dir` is assigned to but never used

Remove assignment to unused variable `script_dir`

(F841)